frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelTags/FeatureOverviewSidePanelTags.tsx

```diff
@@ -0,0 +1,142 @@
+import { IFeatureToggle } from 'interfaces/featureToggle';
+import { useContext, useState } from 'react';
+import { Button, Chip, Divider, styled } from '@mui/material';
+import useTags from 'hooks/api/getters/useTags/useTags';
+import { Add, Cancel } from '@mui/icons-material';
+import AddTagDialog from 'component/feature/FeatureView/FeatureOverview/AddTagDialog/AddTagDialog';
+import { UPDATE_FEATURE } from 'component/providers/AccessProvider/permissions';
+import AccessContext from 'contexts/AccessContext';
+import { Dialogue } from 'component/common/Dialogue/Dialogue';
+import { ITag } from 'interfaces/tags';
+import useFeatureApi from 'hooks/api/actions/useFeatureApi/useFeatureApi';
+import useToast from 'hooks/useToast';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+
+const StyledContainer = styled('div')(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    padding: theme.spacing(3),
+}));
+
+const StyledTagContainer = styled('div')(({ theme }) => ({
+    display: 'flex',
+    gap: theme.spacing(1),
+    flexWrap: 'wrap',
+}));
+
+const StyledChip = styled(Chip)(({ theme }) => ({
+    fontSize: theme.fontSizes.smallBody,
+    '.MuiChip-deleteIcon': {
+        color: theme.palette.neutral.main,
+    },
+}));
+
+const StyledDivider = styled(Divider)(({ theme }) => ({
+    margin: theme.spacing(3),
+    borderStyle: 'dashed',
+}));
+
+const StyledButton = styled(Button)(({ theme }) => ({
+    maxWidth: theme.spacing(20),
+    alignSelf: 'center',
+}));
+
+interface IFeatureOverviewSidePanelTagsProps {
+    feature: IFeatureToggle;
+    header: React.ReactNode;
+}
+
+export const FeatureOverviewSidePanelTags = ({
+    feature,
+    header,
+}: IFeatureOverviewSidePanelTagsProps) => {
+    const { tags, refetch } = useTags(feature.name);
+    const { deleteTagFromFeature } = useFeatureApi();
+
+    const [openTagDialog, setOpenTagDialog] = useState(false);
+    const [showDelDialog, setShowDelDialog] = useState(false);
+    const [selectedTag, setSelectedTag] = useState<ITag>();
+
+    const { setToastData, setToastApiError } = useToast();
+    const { hasAccess } = useContext(AccessContext);
+    const canUpdateTags = hasAccess(UPDATE_FEATURE, feature.project);
+
+    const handleDelete = async () => {
+        if (selectedTag) {
```

Comment:
Might be worth moving this to a guard clause since the entire body is in the if

Reply:
Yep, addressed in https://github.com/Unleash/unleash/pull/2488/commits/df42700f022e809a3347ce3cc364571337d20ffc

Reply:
Like a boss